test/integration/shoots/operations/shoot_op_test.go

```diff
@@ -0,0 +1,128 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package operations_test
+
+import (
+	"context"
+	"flag"
+	"strings"
+	"sync"
+	"time"
+
+	"k8s.io/apimachinery/pkg/labels"
+
+	"github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+	"github.com/gardener/gardener/pkg/logger"
+	. "github.com/gardener/gardener/test/integration/framework"
+	. "github.com/gardener/gardener/test/integration/shoots"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	kubeconfig        = flag.String("kubeconfig", "", "the path to the kubeconfig  of Garden the cluster that will be used for integration tests")
+	shootTestYamlPath = flag.String("shootpath", "", "the path to the shoot yaml that will be used for testing")
+	testShootsPrefix  = flag.String("prefix", "", "prefix to use for test shoots")
+	shootNamespace    = flag.String("shootNamespace", "", "shootNamespace is used for creation and deletion of shoots")
+	logLevel          = flag.String("verbose", "", "verbosity level, when set, logging level will be DEBUG")
+	cleanup           = flag.Bool("cleanup", false, "cleans up all created test shoots after test")
+)
+
+const (
+	WaitForShootCreationTimeout = 3600 * time.Second
+	WaitForShootDeletionTimeout = 3600 * time.Second
+	WaitForCreateDeleteTimeout  = 7200 * time.Second
+)
+
+var _ = Describe("Shoot creation/deletion testing", func() {
+	var (
+		shootGardenerTest         *ShootGardenerTest
+		shoot                     *v1beta1.Shoot
+		shootOperationsTestLogger *logrus.Logger
+	)
+
+	BeforeSuite(func() {
+		if len(*kubeconfig) == 0 {
+			Fail("The path to the Garden cluster kubeconfig is required, please specify it and try again")
+		}
+
+		// parse shoot yaml into shoot object and generate random test names for shoots
+		_, shootObject, err := CreateShootTestArtifacts(*shootTestYamlPath, *testShootsPrefix)
+		Expect(err).To(BeNil())
+
+		shoot = shootObject
+		shootOperationsTestLogger = logger.AddWriter(logger.NewLogger(*logLevel), GinkgoWriter)
+
+		shootGardenerTest, err = NewShootGardenerTest(*kubeconfig, shoot, shootOperationsTestLogger)
+		Expect(err).To(BeNil())
+	})
+
+	AfterSuite(func() {
+		if *cleanup {
+			if len(*shootNamespace) == 0 {
+				Fail(ErrShootsNamespaceWasNotSpecified.Error())
+			}
+
+			shootGardenerTest.Logger.Info("Cleaning up all test shoots!")
+			shoots, err := shootGardenerTest.ShootLister.Shoots(*shootNamespace).List(labels.Everything())
+			Expect(err).To(BeNil())
+
+			var wg sync.WaitGroup
+			for _, shootToDelete := range shoots {
+				if strings.HasPrefix(shootToDelete.Name, IntegrationTestPrefix) {
+					wg.Add(1)
+					go func(shootDel v1beta1.Shoot) {
+						defer wg.Done()
+						shootGardenerCleanup, err := NewShootGardenerTest(*kubeconfig, &shootDel, shootOperationsTestLogger)
```

Comment:
It's a bit of an overkill to create a `NewShootGardenerTest` in order to delete a shoot - why don't you have a well separated method for that?